packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -4,6 +4,320 @@
+      // Firefox.
+      match = lines[0].match(/[<][@](.+):\d+:\d+$/);
+      if (match) return match[1];
+    }
+    // Safari.
+    return lines[0].match(/[@](.+):\d+:\d+$/)[1];
+  }
+  var _url = lookupUrl();
+  var lastSlash = _url.lastIndexOf('/');
+  if (lastSlash == -1) return _url;
+  var currentDirectory = _url.substring(0, lastSlash + 1);
+  return currentDirectory;
+})();
+""";
+
+/// Used to load prerequisite scripts such as ddc_module_loader.js
+const String _simpleLoaderScript = r'''
+window.$dartCreateScript = (function() {
+  // Find the nonce value. (Note, this is only computed once.)
+  var scripts = Array.from(document.getElementsByTagName("script"));
+  var nonce;
+  scripts.some(
+      script => (nonce = script.nonce || script.getAttribute("nonce")));
+  // If present, return a closure that automatically appends the nonce.
+  if (nonce) {
+    return function() {
+      var script = document.createElement("script");
+      script.nonce = nonce;
+      return script;
+    };
+  } else {
+    return function() {
+      return document.createElement("script");
+    };
+  }
+})();
+
+// Loads a module [relativeUrl] relative to [root].
+//
+// If not specified, [root] defaults to the directory serving the main app.
+var forceLoadModule = function (relativeUrl, root) {
+  var actualRoot = root ?? _currentDirectory;
+  return new Promise(function(resolve, reject) {
+    var script = self.$dartCreateScript();
+    let policy = {
+      createScriptURL: function(src) {return src;}
+    };
+    if (self.trustedTypes && self.trustedTypes.createPolicy) {
+      policy = self.trustedTypes.createPolicy('dartDdcModuleUrl', policy);
+    }
+    script.onload = resolve;
+    script.onerror = reject;
+    script.src = policy.createScriptURL(actualRoot + relativeUrl);
+    document.head.appendChild(script);
+  });
+};
+''';
+
+String generateDDCBootstrapScript({
+  required String entrypoint,
+  required String ddcModuleLoaderUrl,
+  required String mapperUrl,
+  required bool generateLoadingIndicator,
+}) {
+  return '''
+${generateLoadingIndicator ? _generateLoadingIndicator() : ""}
+$_currentDirectoryScript
+$_simpleLoaderScript
+
+// A map containing the URLs for the bootstrap scripts in debug.
+let _scriptUrls = {
+  "mapper": "$mapperUrl",
+  "moduleLoader": "$ddcModuleLoaderUrl"
+};
+
+// Create a TrustedTypes policy so we can attach Scripts...
+let _ttPolicy;
+if (window.trustedTypes) {
+  _ttPolicy = trustedTypes.createPolicy("flutter-tools-bootstrap", {
+    createScriptURL: (url) => {
+      let scriptUrl = _scriptUrls[url];
+      if (!scriptUrl) {
+        console.error("Unknown Flutter Web bootstrap resource!", url);
+      }
+      return scriptUrl;
+    }
+  });
+}
+
+// Creates a TrustedScriptURL for a given `scriptName`.
+// See `_scriptUrls` and `_ttPolicy` above.
+function getTTScriptUrl(scriptName) {
+  let defaultUrl = _scriptUrls[scriptName];
+  return _ttPolicy ? _ttPolicy.createScriptURL(scriptName) : defaultUrl;
+}
```

Comment:
It seems the `forceLoadModule` is already creating its own trusted types validation. Do we need this one here as well?

Reply:
(Answer: no, it seems we don't need anything of this stuff, between 90 and 115)

Reply:
Updated!